pkg/crypto/protocol/relay_difficulty.go

```diff
@@ -33,64 +33,54 @@ func IsRelayVolumeApplicable(relayHash, targetHash []byte) bool {
 // on the target number of relays we want the network to mine and the new EMA of
 // the number of relays.
 func ComputeNewDifficultyTargetHash(prevTargetHash []byte, targetNumRelays, newRelaysEma uint64) []byte {
-	// TODO_MAINNET(@red-0ne): Use a language agnostic float implementation to ensure
-	// deterministic results and avoid loss of precision. Specifically, we need to
-	// use big.Rat, delay any computation.
+	// If targetNumRelays == newRelaysEma -> do not scale -> keep the same difficulty to mine relays
+	// TODO_IMPROVE: Figure out if there's a range (e.g. 5%) within which it is reasonable
+	// to keep the same difficulty.
+	if targetNumRelays == newRelaysEma {
+		return prevTargetHash
+	}
 
 	// Calculate the proportion of target relays relative to the EMA of actual volume applicable relays
-	difficultyScalingRatio := new(big.Float).Quo(
-		new(big.Float).SetUint64(targetNumRelays),
-		new(big.Float).SetUint64(newRelaysEma),
-	)
 	// If difficultyScalingRatio < 1 -> scale down -> increase difficulty to mine relays
 	// If difficultyScalingRatio > 1 -> scale up -> decrease difficulty to mine relays
-	if difficultyScalingRatio.Cmp(big.NewFloat(1)) == 0 {
-		return prevTargetHash
-	}
-
-	// You can't scale the base relay difficulty hash below BaseRelayDifficultyHashBz
-	isDecreasingDifficulty := difficultyScalingRatio.Cmp(big.NewFloat(1)) == 1
-	if isDecreasingDifficulty && bytes.Equal(prevTargetHash, BaseRelayDifficultyHashBz) {
+	difficultyScalingRatio := big.NewRat(int64(targetNumRelays), int64(newRelaysEma))
+	scaledDifficultyHashBz := ScaleRelayDifficultyHash(prevTargetHash, difficultyScalingRatio)
+	// Trim any extra zeros from the scaled hash to ensure that it only contains
```

Comment:
```suggestion
	// Trim any leading extra zeros from the scaled hash to ensure that it only contains
```

Reply:
This means we're using little endian, right?

Can you confirm / verify and #PUC